provider/oci/environ.go

```diff
@@ -637,6 +637,17 @@ func (e *Environ) StartInstance(ctx envcontext.ProviderCallContext, args environ
 	}
 	logger.Infof("started instance %q", *machineId)
 
+	var displayName string
+	if machineId == nil {
```

Comment:
we know that when we get here, machineId is not nil so no need for a check

Reply:
I've pulled this functionality out into its own function. IIRC I still include a defensive check.